### PR TITLE
feat: custom starter file

### PR DESCRIPTION
**Pull request by doemser**
Custom-file feature! This will provide more flexibility for users to customize the code generation process to better suit their needs.

### CODE REVIEW

File: base.js [hunk 1]
```diff
@@ -87,10 +88,9 @@ GOAL: ${flags.goal}${
 }
 
 RULES:
-The code should ALWAYS be EXTENDED or FIXED
 The GOAL must be completed
-increment the generation constant ONCE per generation
-EXTEND the CHANGELOG
```
Comment (TimPietrusky) [inline base.js:93]: Will it extend the changelog for the default case still when this rule is removed?

File: base.js [hunk 2]
```diff
@@ -182,7 +188,7 @@ export async function evolve(generation) {
 
 		// Code should start with a comment (changelog). If it doesn't it is often not JavaScrip but
 		// a human language response
-		if (cleanContent.startsWith("/*")) {
+		if (cleanContent.startsWith("/*") || cleanContent.startsWith("import")) {
```
Comment (TimPietrusky) [inline base.js:191]: The whole check was about finding the default case (changelog at the top) because in some cases it was just plain JS. So this check needs to be changed, as it introduces an error case for the default behavior.

**Review by TimPietrusky — CHANGES_REQUESTED** [review]
I wonder what the difference is here besides the missing imports? So you want I have an option to remove imports? Because if this is true the user could also just get rid if them in generation-000.js. As the new custom-000.js is also part of the repo, it would need manual updates too if the user wants to get the latest updates merged.

So please let me know what you think!

**Comment by pixelass** [comment]
Tldr;
Paste into chatGPT.

"Explain this in 2 sentences ..." 😜

Idea:

It was impossible to remove the changelog with any other technique. It happened while trying new prompts. "Surprize me, generative art" would have new ideas in each iteration but the changelog eats tokes (e.g. after 40 iterations). We tried a new flag to remove the changelog, without success. The custom file allows several starters, e.g. one with noise and tone, one with threejs, one with p5 and so on, opening even more flexibility without having to change the content of the files. In a way this is a big addition to quickly switch from starter A to starter B. 

We discussed this on a pairing session and even thought it might even be better to swap the content of both files.

Generation-000: no imports
Custom-000: with noose ans tone and changelog
Custom-p5: p5 starter
Custom-three: thteejs starter
Custom-foo: bar baz

And so on. All these files can be there in parallel while all custom-* are ignored by git except custom-000.

File: base.js (hunk 1 above)
Comment (pixelass) [reply base.js:93]: yes, we tested, please confirm locally.

File: base.js (hunk 2 above)
Comment (pixelass) [reply base.js:191]: On it. (using a new function to check validity)

**Comment by TimPietrusky** [comment]
Ok I see, but then we should make sure, that this makes actually sense for the user. 

My idea in this direction: Having default templates that can be used in the cli, but the entry point generation-000.js stays the same, it will just load the different template. There would be some default templates and people could define their own. Making it easy to extend everything without touching files that are part of the repo. 

We also have to take into account, that the stuff inside project must change based on the starter template that was used, because for Canvas2D you need other default things than for ThreeJS. And specifying those different things in the `generation` or `custom` or `starter` template makes no sense as this would be tokens we have to send.

**Comment by doemser** [comment]
For my feeling about this, the repo should still be dedicated to Canvas 2D content. 

As a user it made super sense to have a custom-000.js as an alternative starting point, additionally I would even like to have a customizable "prompt config" for things that shouldn't always be in the CLI prompt, like the persona, but without losing the default persona. 

From such "user profiles" I would like a clearer and less error-prone CLI prompt, which is more fun, because as soon as you want to deviate from the default, the usability becomes more difficult.
